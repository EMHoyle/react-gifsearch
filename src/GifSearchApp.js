import React, { useState } from 'react';
import AddCategory from './Components/AddCategory';
import GifGrid from './Components/GifGrid';

const GifSearchApp = ( { defaultCategories = []} ) => {
    
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
        <h1>GifSearchApp</h1>
        <AddCategory setCategories={setCategories}/>
        <hr></hr>

        <ul>
            { 
                categories.map( category => 
                    <GifGrid
                        key = {category}
                        category = {category}
                    />
                )
            }
        </ul>
        </>
    )
}

export default GifSearchApp
