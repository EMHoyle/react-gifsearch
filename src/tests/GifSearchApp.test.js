import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifSearchApp from '../GifSearchApp';



describe('GifSearchApp Test', () => {
    
    test('Test 1 - Enzyme: GifSearchApp should be render correctly ', () => {
        
        const wrapper = shallow( <GifSearchApp /> );

        expect(wrapper).toMatchSnapshot();

    });

    test('Test 2 - Enzyme: GifSearchApp shoukd be render categories list', () => {
        
        const categories = ['Friends', 'Umbrella Academy'];
        const wrapper = shallow( <GifSearchApp defaultCategories= {categories} /> );

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });

});