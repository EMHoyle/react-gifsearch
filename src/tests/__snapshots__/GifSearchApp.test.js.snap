// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifSearchApp Test Test 1 - Enzyme: GifSearchApp should be render correctly  1`] = `
<Fragment>
  <h1>
    GifSearchApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`GifSearchApp Test Test 2 - Enzyme: GifSearchApp shoukd be render categories list 1`] = `
<Fragment>
  <h1>
    GifSearchApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Friends"
      key="Friends"
    />
    <GifGrid
      category="Umbrella Academy"
      key="Umbrella Academy"
    />
  </ul>
</Fragment>
`;
